### PR TITLE
[formrecognizer] Update samples

### PR DESCRIPTION
**Pull request by catalinaperalta**
- Update remaining samples that need to use build_mode. 
- Update samples README with new samples.
- Update receipt samples to use doc_type.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2-beta/async_samples/sample_analyze_receipts_async.py [hunk 1]
```diff
@@ -57,11 +57,11 @@ async def analyze_receipts_async():
...
     for idx, receipt in enumerate(receipts.documents):
         print("--------Recognizing receipt #{}--------".format(idx + 1))
-        receipt_type = receipt.fields.get("ReceiptType")
+        receipt_type = receipt.doc_type
         if receipt_type:
             print(
-                "Receipt Type: {} has confidence: {}".format(
-                    receipt_type.value, receipt_type.confidence
+                "Receipt Type: {}".format(
+                    receipt_type.value
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2-beta/async_samples/sample_analyze_receipts_async.py:64]: Oh cool, didn't realize docType would replace ReceiptType like that. Should we simplify and remove the conditional?

`print("Receipt type: {}".format(receipt.doc_type or "N/A"))`

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2-beta/async_samples/sample_analyze_receipts_async.py (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2-beta/async_samples/sample_analyze_receipts_async.py]: I definitely want to confirm this with the rest of the crew, but the receipt schema doesnt include ReceiptType anymore, hence the switch to docType. Simplified the condition!

**Pull request merged**